File: c/structure/holiday.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 10

typedef struct date
{
    int dd;
    int mm;
    int yyyy;
} DATE;

typedef struct holiday
{
    char name[50];
    DATE date;
    char description[100];
} HOLIDAY;

void newHoliday(HOLIDAY *holiday);
void displayHoliday(HOLIDAY *holiday);

int main(int argc, char const *argv[])
{
    HOLIDAY holiday;
    newHoliday(&holiday);
    printf("-------------------------------\n");
    displayHoliday(&holiday);
    return 0;
}

void newHoliday(HOLIDAY *holiday)
{
    printf("Enter holiday name: ");
    scanf("%[^\n]", holiday->name);
    printf("Enter date (dd/mm/yyyy): ");
    scanf("%d/%d/%d%*c", &holiday->date.dd, &holiday->date.mm, &holiday->date.yyyy);
    printf("Enter description: ");
    scanf("%[^\n]", holiday->description);
}

void displayHoliday(HOLIDAY *holiday)
{
    printf("Holiday Name: %s\n", holiday->name);
    printf("Date: %d/%d/%d\n",holiday->date.dd, holiday->date.mm, holiday->date.yyyy);
    printf("Description: %s\n", holiday->description);
}

File: java/no-package/classobject/Holiday.java

import java.time.LocalDate;
import java.util.Scanner;

class Holiday {
    String name;
    LocalDate date;
    String description;

    void newHoliday() {
        Scanner s = new Scanner(System.in);
        System.out.print("Enter holiday name: ");
        name = s.nextLine();
        // System.out.print("Enter date (yyyy-mm-dd): ");
        // String d = s.next();
        // int year = Integer.parseInt(d.substring(0, 3));
        // int month = Integer.parseInt(d.substring(5, 6));
        // int dayOfMonth = Integer.parseInt(d.substring(8, 9));
        int year = 2025;
        int month = 4;
        int dayOfMonth = 6;
        date = LocalDate.of(year, month, dayOfMonth);
        System.out.print("Enter description: ");
        description = s.nextLine();
    }

    void displayHoliday() {
        System.out.println("Holiday Name: " + name);
        System.out.println("Date: " + date);
        System.out.println("Description: " + description);
    }
    public static void main(String[] args) {
        Holiday holiday = new Holiday();
        holiday.newHoliday();
        System.out.println("----------------------------------");
        holiday.displayHoliday();
    }
}